src/ts/domain/UserManager.ts

```diff
@@ -0,0 +1,58 @@
+import { SERVER_ENTRY, API_QUERY, STORAGE_ID } from '../constants';
+import UserType from '../type/UserType';
+import { $ } from '../utils';
+import { alertSnackBar } from '../snackbar';
+
+export interface UserManagerInterface {
+  registerUser(userInfo: UserType): Promise<boolean>;
+  login(loginInfo: { email: string; password: string }): Promise<boolean>;
+  logout(): void;
+}
+
+class UserManager implements UserManagerInterface {
+  registerUser = async ({ email, name, password }) => {
+    try {
+      const response = await fetch(`${SERVER_ENTRY}${API_QUERY.REGISTER}`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'Application/json' },
+        body: JSON.stringify({ email, name, password }),
+      });
+      const result = await response.json();
+      if (!response.ok) {
+        throw new Error(result);
+      }
+    } catch (error) {
+      alertSnackBar(error.message);
+      return Promise.resolve(false);
+    }
+    return Promise.resolve(true);
+  };
+
+  login = async ({ email, password }) => {
+    try {
+      const response = await fetch(`${SERVER_ENTRY}${API_QUERY.LOGIN}`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'Application/json' },
+        body: JSON.stringify({ email, password }),
+      });
+      const result = await response.json();
+      if (!response.ok) {
+        throw new Error(result);
+      }
```

Comment:
```suggestion
      if (!response.ok) {
        throw new Error(result);
      }
      const result = response.json();
```
조건 체크 뒤 값을 가져오는게 좋겠죠? 그리고 여기서 await는 안써도 될 것 같아요.

Reply:
```suggestion
      if (!response.ok) {
        throw new Error(result);
      }
      const result = await response.json();
      localStorage.setItem(STORAGE_ID.LOGIN, result.accessToken);
      
      alertSnackBar(SUCCESS_MESSAGE.LOGIN);
      location.href = 'http://localhost:9000/#!/purchase-product';
```
RegisterView 클래스에서 아래 리턴한 boolean값으로 처리되는 코드가 있는데요. 여기에서 같이 처리할 수 있지 않을까요?
registerUser 메소드도 마찬가지 입니다~ 🙂

Reply:
값을 먼저 검사하고 할당하는 것이 맞네요ㅜ 기초적인 부분인데 고치겠습니다!

Reply:
그런데 await 키워드를 사용하지 않으면 
```Argument of type 'Promise<any>' is not assignable to parameter of type 'string'.```
TS 에러가 뜹니다 ㅠ

Reply:
그리고 다시 코드를 수정하다보니 `throw new Error(result)` 부분에서 result의 에러 메세지를 사용자에게 전달해주기 때문에 결과를 먼저 받고 나서 체크를 하는 기존 방식이 저는 더 맞게 느껴집니다 ㅜㅜ

Reply:
아.. 이건 제가 잘못 리뷰드렸네요. 🥲 에러 객체에 result를 넣는것도 못봤네요. 어휴 죄송합니다 😦...